src/averageAges.js

```diff
@@ -55,7 +84,39 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const onlyWomen = people.filter(person => person.sex === 'f');
+
+  const womenWithChildren = onlyWomen.filter(women => {
+    return people.some(person => person.mother === women.name);
+  });
+
+  let childrenCount = 0;
+
+  const totalAgeDiff = womenWithChildren.reduce((acc, women) => {
+    const ageDiff = ageDiffOfChildren(women);
+
+    return acc + ageDiff;
+  }, 0);
+
+  function ageDiffOfChildren(women) {
+    const children = people.filter(person => {
+      return onlyWithSon
+        ? person.mother === women.name && person.sex === 'm'
+        : person.mother === women.name;
+    });
```

Comment:
just move it out of `ageDiffOfChildren` and you will not be needed to use `childrenCount`  variable, you would use `children.length` instead